crates/marketplace/src/testing/integration.rs

```diff
@@ -78,93 +67,34 @@ where
     /// [`BuilderTask`] it returns will be injected into consensus runtime by HotShot testing harness and
     /// will forward transactions from hotshot event stream to the builder.
     async fn start(
-        n_nodes: usize,
+        _n_nodes: usize,
         url: Url,
         config: Self::Config,
         _changes: HashMap<u64, BuilderChange>,
     ) -> Box<dyn BuilderTask<Types>> {
-        let instance_state = Types::InstanceState::default();
-        let (validated_state, _) = Types::ValidatedState::genesis(&instance_state);
-
         let builder_key_pair = Types::BuilderSignatureKey::generated_from_seed_indexed([0; 32], 0);
 
-        let (senders, receivers) = crate::service::broadcast_channels(BUILDER_CHANNEL_CAPACITY);
-
-        // builder api request channel
-        let (req_sender, req_receiver) = async_broadcast::broadcast::<_>(BUILDER_CHANNEL_CAPACITY);
-
-        let (genesis_payload, genesis_ns_table) =
-            Types::BlockPayload::from_transactions([], &validated_state, &instance_state)
-                .await
-                .expect("genesis payload construction failed");
-
-        let builder_commitment = genesis_payload.builder_commitment(&genesis_ns_table);
-
-        let vid_commitment = {
-            let payload_bytes = genesis_payload.encode();
-            vid_commitment(&payload_bytes, n_nodes)
-        };
-
-        // create the global state
-        let global_state: GlobalState<Types> = GlobalState::<Types>::new(
-            req_sender,
-            senders.transactions.clone(),
-            vid_commitment,
-            Types::View::genesis(),
-        );
-
-        let global_state = Arc::new(RwLock::new(global_state));
-
-        let leaf = Leaf::genesis(&validated_state, &instance_state).await;
-
-        let builder_state = BuilderState::<Types>::new(
-            ParentBlockReferences {
-                view_number: Types::View::genesis(),
-                vid_commitment,
-                leaf_commit: leaf
-                    .commit(&UpgradeLock::<Types, TestVersions>::new())
-                    .await,
-                builder_commitment,
-            },
-            &receivers,
-            req_receiver,
-            Vec::new(), /* tx_queue */
-            Arc::clone(&global_state),
-            Duration::from_millis(1),
-            10,
-            Arc::new(instance_state),
+        let service = GlobalState::new(
+            builder_key_pair,
+            Duration::from_millis(500),
+            Duration::from_millis(10),
             Duration::from_secs(60),
-            Arc::new(validated_state),
+            BUILDER_CHANNEL_CAPACITY,
+            1, // Arbitrary base fee
+            config.hooks,
         );
 
-        builder_state.event_loop();
-
-        let hooks = Arc::new(NoHooks(PhantomData));
-
         // create the proxy global state it will server the builder apis
```

Comment:
Nit: After the test update, `app` is now built from a global state, so not a _proxy_ global state anymore.

Reply:
Yeah, this is a poorly-worded comment. Changed.